src/tools/tidy/src/ui_tests.rs

```diff
@@ -184,7 +185,7 @@ pub fn check(path: &Path, bless: bool, bad: &mut bool) {
 */
 [
 "#;
-        let tidy_src = std::env::current_dir().unwrap().join("src/tools/tidy/src");
+        let tidy_src = PathBuf::from(env::args_os().nth(1).unwrap()).join("src/tools/tidy/src");
```

Comment:
This solution is really unnecessarily cryptic, imo. A better solution would be to pass in the `root_path` here rather than the `tests_path`:

https://github.com/rust-lang/rust/blob/a655e648a9f94d74263108366b83e677af56e35d/src/tools/tidy/src/main.rs#L105

And then adjust the rest of this file to operate from the root, rather than the test dir.